src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -143,8 +143,16 @@ class RedownloadService[F[_]: NonEmptyParallel](
       responses <- apiClients.traverse { client =>
         fetchCreatedSnapshots(client).map(client.id -> _)
       }
+      newProposals = responses.toMap
       proposals <- peersProposals.modify { m =>
-        val updated = m ++ responses.toMap
+        val updated = m.map { case (id, proposalsAtHeight) =>
+          val diff = newProposals
+            .getOrElse(id, Map.empty)
+            .filterKeys(height => !proposalsAtHeight.keySet.contains(height))
```

Comment:
Alternatively a hipster way of diffing Maps:

```
val diff = newProposals.getOrElse(id, Map.empty) -- proposalsAtHeight.keySet
```

As it looks imo cleaner when you use `++` with `--` instead of `++` with `filterKeys`